tests/unit_tests/long_term_block_weight.cpp

```diff
@@ -294,9 +293,11 @@ TEST(long_term_block_weight, pop_invariant_random)
 {
   PREFIX(10);
 
-  for (uint64_t h = 1; h < TEST_LONG_TERM_BLOCK_WEIGHT_WINDOW - 10; ++h)
+  for (uint64_t h = 1; h < 2 * TEST_LONG_TERM_BLOCK_WEIGHT_WINDOW - 10; ++h)
   {
-    size_t w = bc->get_db().height() < TEST_LONG_TERM_BLOCK_WEIGHT_WINDOW ? CRYPTONOTE_BLOCK_GRANTED_FULL_REWARD_ZONE_V5 : bc->get_current_cumulative_block_weight_limit();
+    lcg_seed = bc->get_db().height();
+    uint32_t r = lcg();
+    size_t w = bc->get_db().height() < TEST_LONG_TERM_BLOCK_WEIGHT_WINDOW ? CRYPTONOTE_BLOCK_GRANTED_FULL_REWARD_ZONE_V5 : (r % bc->get_current_cumulative_block_weight_limit());
```

Comment:
The intent is to use the same deterministic block weights used later in the add phase. I think this is needed if the initialization phase generates more than `TEST_LONG_TERM_BLOCK_WEIGHT_WINDOW` blocks, otherwise the above line could've simply been 
```c++
     size_t w = CRYPTONOTE_BLOCK_GRANTED_FULL_REWARD_ZONE_V5;
```
I thought it'd be more useful for this test to use a larger number of initial blocks.

Reply:
OK. I made all the adds use the same line, because it gets copied around in cases where the height maybe can or cannot be < window, so I thought it safer even when the code can't be reached in practice. Anyway, looks good, thanks.